File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        st = set()
        while n != 1:
            if n in st:
                return False
            else:
                st.add(n)
            n = self.sumOfDigits(n)

        return True

    def sumOfDigits(self, n: int) -> int:
        ans = 0
        while n > 0:
            digit = (n % 10) ** 2
            ans += digit
            n //= 10
        return ans

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp

   class Solution {
public:
    
vector<vector<string>> groupAnagrams(vector<string>& strs) {
       
       unordered_map<string , vector<string>> mp;


       for (int i = 0 ; i<strs.size() ; i++){
              vector<int>count(26 , 0);
              for (auto ch : strs[i]){
                     count[ch - 'a']++;
              }
              string key = to_string(count[0]);
              for (auto num : count){
                     key+=','+to_string(num);
              }
              mp[key].push_back(strs[i]);
       }
        
             vector<vector<string>>ans;

             for(const auto &pair : mp){
               
                ans.push_back(pair.second);

             }
             return ans;

        
}
};

